common/src/main/java/com/wynntils/services/itemfilter/ItemFilter.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter;
+
+import com.wynntils.models.items.WynnItem;
+
+public abstract class ItemFilter {
+    protected String searchString;
+
+    protected ItemFilter(String searchString) {
+        this.searchString = searchString;
+    }
+
+    /**
+     * Prepares the operator with the given search string. Similar to compiling for a regex
+     * @return true if the operator was prepared successfully
+     * @throws InvalidSyntaxException if the search string is not invalid
```

Comment:
What does returning false mean? Or put differently, why have an exception instead of returning false? Or, if having an exception is what you want, why have a return value? (I personally prefer return value over exception in this case)

Reply:
It should throw an exception in order to give information on the cause of the failure to prepare the filter. I don't really see why it should ever return false. I will turn this into a void method.

Reply:
Oh i didn't read this right. I don't really see how a return value is better than throwing an exception since we want to explicitely give the cause of the error ?

Reply:
Well I removed the return value in [0c64aeb](https://github.com/Wynntils/Artemis/pull/1860/commits/0c64aeb7b3d2547fc709bd0f4d43d11fbd057936). If you don't think it was the right thing to do, please tell me.